src/components/TeamGenerator/index.tsx

```diff
@@ -41,70 +82,134 @@ export default function TeamGenerator({ okCallback, numOfTeams, hideModal }: Pro
       teamsPlayers[i].push(validPlayerNames[i]);
     }
 
-    setTeams(
-      teamsPlayers.map((playerNames) => ({
-        teamId: nanoid(),
-        name: getCommaSeparatedStringWithAndBeforeTheLastItem(playerNames),
-        score: 0,
-        selectedOptions: [],
-      })),
+    teamsForm.setValue(
+      'teams',
+      teamsPlayers.map((playerNames) => getCommaSeparatedStringWithAndBeforeTheLastItem(playerNames)).join('\n'),
     );
+    setPlayers(teamsPlayers.map((playerList) => playerList.join(',')));
+    setShouldShowTeams(true);
   }
 
+  function submitTeamNamesForm({ teams }) {
+    createTeams({
+      teams: teams.split('\n').map((name, idx) => ({
+        name,
+        players: players[idx],
+        avatarColor: getRandomColor(),
+      })),
+      players,
+      mode: 'automatic',
+    });
+  }
+
+  const shouldBeMoreThanOne = (value: number) => value >= 2 || 'Should be more than 1';
+  const validateNumberOfTeams = (value: string) =>
+    value.split('\n').length === parseInt(`${teamCount}`) || `Number of teams should not be more than ${teamCount}`;
+
   return (
-    <Modal open size="tiny" centered={false} dimmer="blurring">
-      <Modal.Header>Generate team with player names</Modal.Header>
-      <Modal.Content className="flexCol flex">
-        <div className="flex flexCol">
-          <p className={styles.info}>
-            Enter one player name in each line or paste names from a spreadsheet column (Excel, Google sheet, etc.).
-          </p>
-          <Label as="label" className={styles.playerNames}>
-            <div className="mb-md">Player names</div>
-            <TextArea
-              className="fullWidth"
-              rows={10}
-              value={playerNames}
-              onChange={(ev) => setPlayerNames(ev.target.value)}
+    <>
```

Comment:
Shouldn't this be inside a modal?

Reply:
no.. it just Modal body.. we have a common modal component